### DISCUSSION TITLE
How do I get the power for each Instance？

### DISCUSSION THREAD

**Discussion by LT-HB** [The-OpenROAD-Project/OpenROAD#4417] [Q&A]
How do I get the power for each Instance？I have no power report after reading the odb file in gui.
![power_report](https://github.com/The-OpenROAD-Project/OpenROAD/assets/107295305/d747e889-1ea2-448e-ba0b-11b6c08e2b52)

**Comment by maliberty** [comment]
```report_power -instances [get_cells *]``` should do it.

**Reply by LT-HB** [reply]
Why is there no power information in the ODB file?

**Reply by maliberty** [reply]
Mostly history.  Power and timing are both computed by OpenSTA which has its own data structures that are independent of odb.  Its why in ORFS we have the 'make gui_<step>' targets to reload the timing data as well as the odb.  If you use ORFS then please do 'make gui_final' and see if that fixes report_power.

**Comment by b224hisl** [comment]
can I get the power of each module?

**Reply by b224hisl** [reply]
And I also want to know the relationship among 'cell', 'instance', 'module'. I would like to get the power report of a specific module indexed by its name.

**Reply by LT-HB** [reply]
When I running the flow, I add 

> report_power -instances [get_cells *]

 to my tcl command. The result is shown
![image](https://github.com/The-OpenROAD-Project/OpenROAD/assets/107295305/c50c5428-0503-4f21-ad66-916f6f1985d7)
I use tinyRocket design.

**Reply by b224hisl** [reply]
yes, thank you, but it shows the power of all cells... Do you know is there a command to report the power of a specific module? maybe get_cells of a module can make it

**Comment by maliberty** [comment]
There is no per-module report today.  If you can express what you want in terms of instance names you could do something like `report_power -instances [get_cells a/b/*]`

**Reply by b224hisl** [reply]
ok fine， thank you, maybe I have to write a script to do the adding

**Comment by maliberty** [comment]
get_cells comes from SDC where a 'cell' is another (poor) name for an instance.
